tests/41end-to-end-keys/04-query-key-federation.pl

```diff
@@ -28,8 +28,10 @@
 
          # TODO: Check that the content matches what we uploaded.
 
-         assert_eq( $alice_device_keys->{"unsigned"}->{"device_display_name"},
-                    "test display name" );
+         # Device display names are optional for POST /user/keys/query responses.
+         # If one exists, ensure it's the one we expected.
+         my $device_display_name = $alice_device_keys->{"unsigned"}->{"device_display_name"} // "";
+         assert_eq "test display name", $device_display_name, "device display name";
```

Comment:
Errr on second thoughts: is this correct? if the `device_display_name` field is missing we'll then `assert_eq "test display name" "" "device_display_name"`, which will fail the test?

Reply:
Oop, shows me right for blindly copying. Have hopefully fixed in https://github.com/matrix-org/sytest/pull/1311/commits/c19f460d99107d0322f925230f644493880c85ef.